python/tvm/relay/frontend/tensorflow.py

```diff
@@ -1605,9 +1605,12 @@ def _while_loop(self):
         loop_vars = []
         bind_map = {}
         for i, var in enumerate(self.loop_vars):
-            assert isinstance(var, _expr.Var), repr(var)
-            v = tvm.relay.var("loop_var" + str(i),
-                              type_annotation=var.type_annotation)
+            if not isinstance(var, _expr.Var):
+                var_type = ir_pass.infer_type(var).checked_type
+            else:
+                var_type = var.type_annotation
+
+            v = tvm.relay.var("loop_var" + str(i), type_annotation=var_type)
             loop_vars.append(v)
             bind_map[var] = v
```

Comment:
I think instead of calling bind which has specific semantics we should just add code like this:

```
class RewriteSubgraph(ExprMutator):
    def __init__(self, rewrite_map): 
        self.rewrite_map = rewrite_map

    def visit(self, expr):
       if expr in self.rewrite_map:
            return self.rewrite_map[expr]
       else: 
            super().visit(expr)

def rewrite_subgraph(expr, rewrites):
     return RewriteSubgraph(rewrites).visit(expr)
```
then replace the call with bind to this, this will handle the general case and not just calls.

Reply:
Thanks. Do you think it should be a common function or just be kept in TF frontend?

Reply:
I think it is reasonable to just put it in the TF pass for now. We could also put in the frontend/common file where utilities are kept.

Reply:
I've modified it according to request.